exporters/otlp/include/opentelemetry/exporters/otlp/recordable.h

```diff
@@ -21,7 +21,13 @@ class Recordable final : public sdk::trace::Recordable
   void SetAttribute(nostd::string_view key,
                     const opentelemetry::common::AttributeValue &value) noexcept override;
 
-  void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept override;
+  void AddEvent(nostd::string_view name,
+                core::SystemTimestamp timestamp,
+                const trace::KeyValueIterable &attributes) noexcept override;
+
+  void AddLink(opentelemetry::trace::TraceId trace_id,
+               opentelemetry::trace::SpanId span_id,
```

Comment:
Should we rather pass a `SpanContext` here? It holds both trace id, span id and trace state.

Reply:
The spec actually asked for `SpanContext` https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#add-links.

Reply:
Are we going to also be setting the trace state here?

Reply:
https://github.com/open-telemetry/opentelemetry-proto/blob/c44400dc2b039380383b4fa4f243c49d7baf83c1/opentelemetry/proto/trace/v1/trace.proto#L196